### ISSUE TITLE
Suport rr as a variant of gdb

### ISSUE THREAD

**Issue by MaskRay** [realgud/realgud#224]
[rr](https://rr-project.org/) records execution trace and provides efficient reverse execution. The `realgud:gdb` command can be reused (let the user input `rr replay`) but some process/prompt detection needs changes (prompt changes from `(gdb)` to `(rr)`, etc)

**Comment by rocky** [comment]
I don't think we want to add any more debuggers into realgud proper. (In fact, I might consider removing some.) 

So what you want do is follow [wiki instructions for adding a new debugger](https://github.com/realgud/realgud/wiki/How-to-add-a-new-debugger). The best way to make sure this happens is to get involved. If you make a serious effort, I'll help.
